app/Helpers/database/set-claim.php

```diff
@@ -8,47 +8,58 @@
 use App\Community\Enums\ClaimType;
 use App\Enums\Permissions;
 use App\Models\AchievementSetClaim;
+use App\Models\User;
 use App\Support\Cache\CacheKey;
 use Carbon\Carbon;
 use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Cache;
 use Illuminate\Support\Facades\DB;
 
 /**
  * Checks if the user is able to make a claim and inserts the claim into the database. If the claim
  * is a collaboration claim then the claim will have the same Finished time as the primary claim for the game.
  */
-function insertClaim(string $user, int $gameID, int $claimType, int $setType, int $special, int $permissions): bool
+function insertClaim(User $user, int $gameId, int $claimType, int $setType, int $special): bool
 {
-    if ($claimType === ClaimType::Primary) {
-        // Primary Claim
-        // Prevent if user has no available slots except for when they are the sole dev of the set
-        if ($special === ClaimSpecial::None && getActiveClaimCount($user, false) >= permissionsToClaim($permissions)) {
-            return false;
-        }
+    $userPermissions = (int) $user->getAttribute('Permissions');
 
-        $now = Carbon::now();
-        $finished = Carbon::now()->addMonths(3);
+    $finishedAt = Carbon::now()->addMonths(3); // Default to 3 months from now.
 
-        $query = "
-            INSERT INTO
-                SetClaim (`User`, `GameID`, `ClaimType`, `SetType`, `Status`, `Extension`, `Special`, `Created`, `Finished` ,`Updated`)
-            VALUES
-                ('$user', '$gameID', '$claimType', '$setType', '" . ClaimStatus::Active . "', '0', '$special', '$now', '$finished', '$now')";
+    if ($claimType !== ClaimType::Primary) {
+        $primaryClaimFinishTime = AchievementSetClaim::where('game_id', $gameId)
+            ->where('Status', ClaimStatus::Active)
+            ->where('ClaimType', ClaimType::Primary)
+            ->value('Finished');
 
-        return legacyDbStatement($query);
+        // If a primary claim doesn't exist, treat the new claim as a primary claim.
+        if ($primaryClaimFinishTime) {
+            $finishedAt = $primaryClaimFinishTime;
+            $special = ClaimSpecial::None;
+        }
+    } else {
+        // Different roles are allowed a maximum number of active primary claims.
+        // Does this user currently have fewer primary claims than that maximum?
+        $isUserAllowedToClaim = getActiveClaimCount($user->User, false) < permissionsToClaim($userPermissions);
+
+        if ($special === ClaimSpecial::None && !$isUserAllowedToClaim) {
```

Comment:
`getActiveClaimCount` could be avoided in the `$special == ClaimSpecial::None` was moved up to the else (line 39)

Reply:
Good catch, updated in latest.